pkgs/by-name/im/imake/package.nix

```diff
@@ -64,14 +64,14 @@ stdenv.mkDerivation (finalAttrs: {
     CFLAGS = "-DIMAKE_COMPILETIME_CPP='\"${
       if stdenv.hostPlatform.isDarwin then "${tradcpp}/bin/cpp" else "gcc"
     }\"'";
+    inherit tradcpp;
   };
 
   preInstall = ''
     mkdir -p $out/lib/X11/config
     ln -s ${xorg-cf-files}/lib/X11/config/* $out/lib/X11/config
   '';
 
-  inherit tradcpp xorg-cf-files;
   setupHook = ./setup-hook.sh;
```

Comment:
Sorry for the piecemeal feedback.

I think we should not rely on environment variable based substitution here (that's the whole stupidity of `substituteAll`).

Let's do `setupHook = replaceVars ...` to make this replacement explicit. That way we don't need to set `tradcpp` as an environment variable, but can just pass it to replaceVars.

Reply:
No problem, I'm just glad to get feedback and merges instead of having a pile of PRs lingering for ages :)

Changes implemented.